astropy/table/table.py

```diff
@@ -2021,6 +2021,70 @@ def group_by(self, keys):
 
         return groups.table_group_by(self, keys)
 
+    def to_pandas(self):
+        """
+        Return a :class:`pandas.DataFrame` instance
+
+        Returns
+        -------
+        dataframe : :class:`pandas.DataFrame`
+            A pandas :class:`pandas.DataFrame` instance
+        """
+        from pandas import DataFrame
+
+        if self.has_mixin_columns:
+            raise ValueError("Cannot convert a table with mixin columns to a pandas DataFrame")
+
+        for column in self.columns:
+            if len(self.dtype[column].shape) > 0:
+                raise ValueError("Cannot convert a table with multi-dimensional columns to a pandas DataFrame")
+
+        if self.masked:
+            # We have to add the columns manually so as to treat the mask
+            # correctly
+            d = DataFrame()
+            for name, column in self.columns.items():
+                if isinstance(column, MaskedColumn):
+                    if column.dtype.kind in ['i']:
+                        d[name] = column.astype(float).filled(np.nan)
+                    elif column.dtype.kind in ['f', 'c']:
+                        d[name] = column.filled(np.nan)
+                    else:
+                        d[name] = column.astype(np.object).filled(np.nan)
+        else:
+            d = DataFrame(self.as_array())
+
+        return d
+
+    @classmethod
+    def from_pandas(cls, dataframe):
+        """
+        Create a `Table` from a :class:`pandas.DataFrame` instance
+
+        Parameters
+        ----------
+        dataframe : :class:`pandas.DataFrame`
+            The pandas :class:`pandas.DataFrame` instance
+
+        Returns
+        -------
+        table : `Table`
+            A `Table` instance
+        """
+
+        self = cls()
```

Comment:
It's annoying to me, but at this moment it is still noticeably faster to collect columns in a dict `out` and then create the table from that object at the end with `cls(out, names=dataframe.columns, copy=False)`